lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -631,6 +634,31 @@ void SpecialFunctionHandler::handleErrnoLocation(
                            executor.kmodule->targetData->getTypeSizeInBits(
                                target->inst->getType())));
 }
+
+void SpecialFunctionHandler::handleNewlocale(ExecutionState &state,
+                                            KInstruction *target,
+                                            std::vector<ref<Expr> > &arguments) {
+#ifdef __APPLE__
+  executor.terminateStateOnError(
+      state, "check_memory_access requires constant args", Executor::User);
+#else
+  int mask =
+      dyn_cast<ConstantExpr>(arguments[0])->getZExtValue(sizeof(int) * 8);
+  locale_t base = (locale_t)dyn_cast<ConstantExpr>(arguments[2])
+                      ->getZExtValue(sizeof(locale_t) * 8);
+  locale_t x =
```

Comment:
The resource handling of the `newlocale` is not done properly in general.
Here you create a leak: `newlocale` is executed as part of the KLEE process and needs to be freed. This won't happen. The `addexternalObject` makes it accessible as a memory object but won't free it.
If you fork/destroy states a lot and inbetween you execute `newlocale` things will get bad.

Beside that, according to documentation of `newlocale`:
> If base refers to valid existing locale object (i.e., an object
          returned by a previous call to newlocale() or duplocale(3)), then
          that object is modified by the call.

Therefore, if you execute this call in one state refering to a previously created local you will change that locale object that is referenced by the otherone.